### PR TITLE
feat: improve js docs

### PR DESCRIPTION
**Pull request by himanshu-dixit**

### CODE REVIEW

File: mint.json [hunk 1]
```diff
@@ -81,14 +81,14 @@
...
       "icon": "terminal",
       "url": "sdk"
     },
-     {
-      "name": "JS SDK Reference",
+    {
+      "name": "API",
       "icon": "code",
-      "url": "https://composiohq.github.io/composio-js/index.html"
+      "url": "api-reference"
```
Comment (himanshu-dixit) [inline mint.json:91]: Reverting to yesterday and removing js sdk reference

File: sdk/javascript/langchain.mdx [hunk 2]
```diff
@@ -5,81 +5,94 @@ icon: "code"
...
 import { ChatOpenAI } from "@langchain/openai";
 import { createOpenAIFunctionsAgent, AgentExecutor } from "langchain/agents";
 import { pull } from "langchain/hub";
-
 import { LangchainToolSet } from "composio-core";
       ```
     </CodeGroup>
 </Step>
-<Step title="Initialize LangchainToolSet and setup agent">
+<Step title="Define method to ask user to login">
```
Comment (kaavee315) [inline sdk/javascript/langchain.mdx:20]:
```suggestion
<Step title="Define method to let users connect their GitHub account">
```

File: sdk/javascript/langchain.mdx [hunk 3]
```diff
@@ -5,81 +5,94 @@ icon: "code"
...
-    tools,
-    prompt,
-    });
-
-    const agentExecutor = new AgentExecutor({
-    agent,
-    tools,
-    verbose: true,
-    });
+      ```typescript  Define method to ask user to login
```
Comment (kaavee315) [inline sdk/javascript/langchain.mdx:23]:
```suggestion
      ```typescript  Define method to let users connect their GitHub account
```

File: sdk/javascript/langchain.mdx [hunk 4]
```diff
@@ -5,81 +5,94 @@ icon: "code"
...
-
+Learn more about Entities [here](introduction/foundations/components/entity/entity-guide)
 </Step>
 
-<Step title="Invoke the agent">
+<Step title="Setup Agent with Langchain">
 
 <CodeGroup>
-      ```typescript Add openAI assistant handler
+      ```typescript Define your Agent
```
Comment (kaavee315) [inline sdk/javascript/langchain.mdx:47]:
```suggestion
      ```typescript Define your Agent to create issues on Github
```

File: sdk/javascript/langchain.mdx [hunk 5]
```diff
@@ -5,81 +5,94 @@ icon: "code"
...
-    });
+      ```typescript  Define method to ask user to login
+const toolset = new LangchainToolSet({ apiKey: process.env.COMPOSIO_API_KEY, });
+
+async function setupUserConnectionIfNotExists(entityId) {
+        const entity = await toolset.client.getEntity(entityId);
+        const connection = await entity.getConnection('github');
+
+        if (!connection) {
+            const connection = await entity.initiateConnection(appName);
```
Comment (kaavee315) [inline sdk/javascript/langchain.mdx:31]:
```suggestion
            // If this entity/user hasn't already connected the account
            const connection = await entity.initiateConnection(appName);
```

File: sdk/javascript/langchain.mdx [hunk 6]
```diff
@@ -5,81 +5,94 @@ icon: "code"
...
 
-    res.json({ output: result.output });
+<CodeGroup>
+      ```typescript  Invoke the agent
+      executeGithubAgent("himanshu")
       ```
     </CodeGroup>
 
-
+Tada🎉! It was this simple to create a powerful Agent with Composio and Langchain.
```
Comment (kaavee315) [inline sdk/javascript/langchain.mdx:90]:
```suggestion
Tada 🎉! It was this simple to create a powerful Agent with Composio and Langchain.
```

File: sdk/javascript/openai.mdx [hunk 7]
```diff
@@ -5,70 +5,86 @@ icon: "code"
...
  <CodeGroup>
       ```javascript Import OpenAI and OpenAIToolSet
-      import { OpenAI } from "openai";
+import { OpenAI } from "openai";
 import { OpenAIToolSet } from "composio-core";
       ```
     </CodeGroup>
 </Step>
-<Step title="Initialize OpenAIToolSet and OpenAI">
+<Step title="Define method to ask user to login">
```
Comment (kaavee315) [inline sdk/javascript/openai.mdx:17]:
```suggestion
<Step title="Define method to let user connect their GitHub account">
```

File: sdk/javascript/openai.mdx [hunk 8]
```diff
@@ -5,70 +5,86 @@ icon: "code"
...
 
  <CodeGroup>
-      ```typescript Import OpenAI and OpenAIToolSet
+      ```typescript Define method to ask user to login
```
Comment (kaavee315) [inline sdk/javascript/openai.mdx:20]:
```suggestion
      ```typescript Define method to let user connect their GitHub account
```

File: sdk/javascript/openai.mdx [hunk 9]
```diff
@@ -5,70 +5,86 @@ icon: "code"
...
 
-const tools = await toolset.get_actions({
-            actions: ["github_issues_create"]
-});
+async function setupUserConnectionIfNotExists(entityId) {
+        const entity = await toolset.client.getEntity(entityId);
+        const connection = await entity.getConnection('github');
 
+        if (!connection) {
+            const connection = await entity.initiateConnection(appName);
```
Comment (kaavee315) [inline sdk/javascript/openai.mdx:30]:
```suggestion
            // If this entity/user hasn't already connected the account
            const connection = await entity.initiateConnection(appName);
```

**Review by kaavee315 — APPROVED** [review]
other than a few suggestions

**Comment by himanshu-dixit** [comment]
Thanks! Merging

**Pull request merged**